Lib/multiprocessing/popen_spawn_win32.py

```diff
@@ -71,7 +71,7 @@ def __init__(self, process_obj):
             # start process
             try:
                 hp, ht, pid, tid = _winapi.CreateProcess(
-                    python_exe, cmd,
+                    os.fsdecode(python_exe), cmd,
```

Comment:
It should be converted to a string in `spawn.set_executable()`. 

Note that the `cmd` string is from `' '.join('"%s"' % x for x in cmd)`, which requires each item of the `cmd` list to support string formatting. The list returned by `spawn.get_command_line()` contains `_python_exe` as the first element. Path-like objects need only define the path for `__fspath__()`, not for string conversion.

Reply:
Thanks @eryksun! And I even think that conversion of `_python_exe` to `str` for Windows and `bytes` for Unix-like systems should happen in `spawn.get_executable()` (achieves information hiding), and therefore that `spawn.get_command_line()` should call `spawn.get_executable()` instead of accessing `_python_exe` directly (violates information hiding), and that `utility.spawnv_passfds()` should not perform the conversion of `_python_exe` to `bytes` anymore (violates information hiding). I have just updated the PR.

Reply:
Well I think you are right that it is also possible to perform the conversion of `spawn._python_exe` in `spawn.set_executable(exe)` since this is the only entry point for a user-provided path (besides rebinding `spawn._python_exe`, but this is not part of the API).